packages/gatsby-dev-cli/src/utils/get-dependant-packages.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Recursively get set of packages that depend on given package.
+ * Set also includes passed package.
+ */
+const getDependantPackages = ({
+  packageName,
+  depTree,
+  packagesToPublish = new Set(),
+}) => {
+  packagesToPublish.add(packageName)
+  const dependants = depTree[packageName]
+  if (dependants) {
+    dependants.forEach(dependant =>
+      getDependantPackages({
```

Comment:
This can cause an infinite loop. You probably want something like this:
```js
const alreadySeen = packagesToPublish.has(packageName);
packagesToPublish.add(packageName)
  const dependants = depTree[packageName]
  if (dependants && !alreadySeen) {
```